int32[3] start_grid_coords_dir
int32[3] goal_grid_coords_dir
int32[] cmd_list